src/exojax/spec/api.py

```diff
@@ -0,0 +1,636 @@
+        """Partition Function ratio using HAPI partition data.
+        (This function works for JAX environment.)
+
+        Args:
+           T: temperature (K)
+
+        Returns:
+           Qr_line, partition function ratio array for lines [Nlines]
+
+        Note:
+           Nlines=len(self.nu_lines)
+        """
+        qrx = []
+        for idx, iso in enumerate(self.uniqiso):
+            qrx.append(self.qr_interp(idx, T))
+
+        qr_line = jnp.zeros(len(self.isoid))
+        for idx, iso in enumerate(self.uniqiso):
+            mask_idx = np.where(self.isoid == iso)
+            qr_line = qr_line.at[jnp.index_exp[mask_idx]].set(qrx[idx])
+        return qr_line
+
+
+MdbHit = MdbHitemp  #compatibility
+
+
+class MdbHitran(HITRANDatabaseManager):
+    """molecular database of HITRAN
+
+    Attributes:
+        simple_molecule_name: simple molecule name
+        nurange: nu range [min,max] (cm-1)
+        nu_lines (nd array): line center (cm-1)
+        Sij0 (nd array): line strength at T=Tref (cm)
+        dev_nu_lines (jnp array): line center in device (cm-1)
+        logsij0 (jnp array): log line strength at T=Tref
+        A (jnp array): Einstein A coeeficient
+        gamma_natural (jnp array): gamma factor of the natural broadening
+        gamma_air (jnp array): gamma factor of air pressure broadening
+        gamma_self (jnp array): gamma factor of self pressure broadening
+        elower (jnp array): the lower state energy (cm-1)
+        gpp (jnp array): statistical weight
+        n_air (jnp array): air temperature exponent
+    """
+    def __init__(self,
+                 path,
+                 nurange=[-np.inf, np.inf],
+                 margin=0.0,
+                 crit=0.,
+                 Ttyp=1000.,
+                 isotope=None,
+                 gpu_transfer=False):
+        """Molecular database for HITRAN/HITEMP form.
+
+        Args:
+           path: path for HITRAN/HITEMP par file
+           nurange: wavenumber range list (cm-1) [min,max] or wavenumber grid
+           margin: margin for nurange (cm-1)
+           crit: line strength lower limit for extraction
+           Ttyp: typical temperature to calculate Sij(T) used in crit
+           isotope: None= use all isotopes. 
+           gpu_transfer: tranfer data to jnp.array?
+        """
+
+        self.path = pathlib.Path(path).expanduser()
+        self.molecid = search_molecid(str(self.path.stem))
+        self.simple_molecule_name = get_molecule(self.molecid)
+
+        #numinf, numtag = hitranapi.read_path(self.path)
+        self.crit = crit
+        self.Ttyp = Ttyp
+        self.margin = margin
+        self.nurange = [np.min(nurange), np.max(nurange)]
+        load_wavenum_min = self.nurange[0] - self.margin
+        load_wavenum_max = self.nurange[1] + self.margin
+
+        super().__init__(
+            molecule=self.simple_molecule_name,
+            name="HITRAN-{molecule}",
+            local_databases=self.path.parent,
```

Comment:
In contrast, rewriting this line to `local_databases="/",` still left `MdbHitran` with a PermissionError as below.

```
from exojax.spec.api import MdbHitran
mdb = MdbHitran(".database/CO/05_hit12.par", nurange=[4200.0, 4300.0], crit=1.e-30) 
```
↓
```
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
/tmp/ipykernel_4125346/607486000.py in <module>
      1 from exojax.spec.api import MdbHitran
----> 2 mdb = MdbHitran(".database/CO/05_hit12.par",
      3                  nurange=[4200.0, 4300.0],
      4                  crit=1.e-30) 

~/miniconda3/lib/python3.9/site-packages/exojax-1.1.3-py3.9.egg/exojax/spec/api.py in __init__(self, path, nurange, margin, crit, Ttyp, isotope, gpu_transfer)
    502         download_files = self.get_missing_files(local_file)
    503         if download_files:
--> 504             self.download_and_parse(download_files,
    505                                     cache=True,
    506                                     parse_quanta=True)

~/miniconda3/lib/python3.9/site-packages/radis-0.13.0-py3.9-linux-x86_64.egg/radis/api/hitranapi.py in download_and_parse(self, local_file, cache, parse_quanta)
   1183         # Use HAPI only to download the files, then we'll parse them with RADIS's
   1184         # parsers, and convert to RADIS's fast HDF5 file formats.
-> 1185         isotope_list, data_file_list, header_file_list = download_all_hitran_isotopes(
   1186             molecule, tempdir
   1187         )

~/miniconda3/lib/python3.9/site-packages/radis-0.13.0-py3.9-linux-x86_64.egg/radis/api/hitranapi.py in download_all_hitran_isotopes(molecule, directory)
   1141             from radis.misc.basics import make_folders
   1142 
-> 1143             make_folders(*split(abspath(directory)))
   1144 
   1145             db_begin(directory)

~/miniconda3/lib/python3.9/site-packages/radis-0.13.0-py3.9-linux-x86_64.egg/radis/misc/basics.py in make_folders(path, folders)
     37         folders = [folders]
     38     for folder in folders:
---> 39         os.makedirs(
     40             join(path, folder), exist_ok=True
     41         )  # makedirs create folders recursively

~/miniconda3/lib/python3.9/os.py in makedirs(name, mode, exist_ok)
    213     if head and tail and not path.exists(head):
    214         try:
--> 215             makedirs(head, exist_ok=exist_ok)
    216         except FileExistsError:
    217             # Defeats race condition when another thread created the path

~/miniconda3/lib/python3.9/os.py in makedirs(name, mode, exist_ok)
    223             return
    224     try:
--> 225         mkdir(name, mode)
    226     except OSError:
    227         # Cannot rely on checking for EEXIST, since the operating system

PermissionError: [Errno 13] Permission denied: '/downloads__can_be_deleted'
```